src/test/java/org/haffson/adventofcode/days/day05/Day05Test.java

```diff
@@ -0,0 +1,164 @@
+
+        //act
+        final String actualResult = day05.firstPart();
+
+        //assert
+        assertThat(actualResult).isEqualTo(expectedResult);
+    }
+
+    @Test
+    void convertStringToCharacterList_returnListOfChars() {
+        //arrange
+        final Day05 day05 = new Day05(fileReaders);
+        final String inputString = "dabAcCaCBAcCcaDA";
+
+//        final List<Character> expectedResult = List.of('d',"a","b","A","c","C","c","a","C","B","A","c","C","c","a","D","A");
+        final List<Character> expectedResult = List.of('d', 'a', 'b', 'A', 'c', 'C', 'a', 'C', 'B', 'A', 'c', 'C', 'c', 'a', 'D', 'A');
+
+        //act
+        final List<Character> actualResult = day05.convertStringToCharacterList(inputString);
+
+        //assert
+        assertThat(actualResult).isEqualTo(expectedResult);
+    }
+
+    @Test
+    void convertStringListToString_returnString() {
+        //arrange
+        final Day05 day05 = new Day05(fileReaders);
+        final List<String> inputStringList = List.of("dabAcCaCBAcCcaDA");
+
+        final String expectedResult = "dabAcCaCBAcCcaDA";
+
+        //act
+        final String actualResult = day05.convertStringListToString(inputStringList);
+
+        //assert
+        assertThat(actualResult).isEqualTo(expectedResult);
+    }
+
+    @Test
+    void removeCharDuplicatesWithDifferentCases_returnListWithoutDuplicatesInDifferentCases() {
+        //arrange
+        final Day05 day05 = new Day05(fileReaders);
+        // new ArrayList because I need a mutable list
+        final List<Character> inputList = new ArrayList<>(List.of('d', 'a', 'b', 'A', 'c', 'C', 'a', 'C', 'B', 'A', 'c', 'C', 'c', 'a', 'D', 'A'));
+
+        final List<Character> expectedResult = List.of('d', 'a', 'b', 'C', 'B', 'A', 'c', 'a', 'D', 'A');
+
+        //act
+        final List<Character> actualResult = day05.removeCharDuplicatesWithDifferentCases(inputList);
+
+        //assert
+        assertThat(actualResult).isEqualTo(expectedResult);
+    }
+
+    @Test
+    void test_secondPart_returnsExpectedResult() {
+        //arrange
+        final Day05 day05 = new Day05(fileReaders);
+        when(fileReaders.getInputList("src/main/resources/puzzle_input/day5_input.txt"))
+                .thenReturn(List.of("dabAcCaCBAcCcaDA"));
+
+        final String expectedResult = "Part 2 - Shortest remaining Units after fully reacting the polymer and removing one letter: " + 4;
+
+        //act
+        final String actualResult = day05.secondPart();
+
+        //assert
+        assertThat(actualResult).isEqualTo(expectedResult);
+    }
+
+    @Test
+    void getPolymerLengthByRemovedChar_returnIntValueOfPolymerLength() {
+        //arrange
+        final Day05 day05 = new Day05(fileReaders);
+        final List<Character> inputList = new ArrayList<>(List.of('d', 'a', 'b', 'A', 'c', 'C', 'a', 'C', 'B', 'A', 'c', 'C', 'c', 'a', 'D', 'A'));
+
+        final Map<Character, Integer> expectedResult = Stream.of(new Object[][]{
```

Comment:
Hier könntest du wahrscheinlich auch `Map.of()` benutzen.

Wenn es zuviele Einträge sind, dann gibt es auch eine Version von `Map.of()`, die `MapEntry`-Objekte nimmt (dafür aber beliebig viele davon).

Reply:
Map.of() darf nur 10 Einträge enthalten.
Habe bei Baeldung dann geschaut, wie man eine Map sonst mit mehr Einträgen initialisieren kann und bin auf verschiedene Varianten gestoßen.
Wenn Map.of() mit MapEntry aber best practice ist, nutze ich das gerne

Reply:
Ist nur eine Option.

Kannst auch eine Map erzeugen und `put`-Aufrufe machen.

Stream von einem Array it Cast fand ich nur ungewöhnlich.

Reply:
.put hätte auch funktioniert
Habe nur streams gelesen und dachte mir " Yeeey streams" :D
Das **Map.ofEntries** gefällt mir ganz gut - danke für den Tipp.